src/Symfony/Component/Messenger/Bridge/Amqp/Transport/AmqpReceiver.php

```diff
@@ -52,6 +52,12 @@ private function getEnvelope(string $queueName): iterable
     {
         try {
             $amqpEnvelope = $this->connection->get($queueName);
+        } catch (\AMQPConnectionException) {
+            // Try to reconnect once to accommodate need for one of the nodes in cluster needing to stop serving the
+            // traffic. This may happen for example when one of the nodes in cluster is going into maintenance node.
+            // see https://github.com/php-amqplib/php-amqplib/issues/1161
+            $this->connection->queue($queueName)->getConnection()->reconnect();
+            $amqpEnvelope = $this->connection->get($queueName);
```

Comment:
What if this line throws an exception? 
I think we must add try/catch around these two new lines too.

Reply:
If I understand correct, here we want the exception as we only wants try to reconnect once?

Reply:
Yeah if it fails second time, exception being thrown this time is fair

Reply:
When it fails, we should convert the `AMQPException` exception to a `TransportException` one, right?

Reply:
Oh I see. Will do.